src/android/nl/xservices/plugins/HeadsetDetection.java

```diff
@@ -37,23 +41,20 @@ public void initialize(CordovaInterface cordova, CordovaWebView webView) {
       mCachedWebView = webView;
       IntentFilter intentFilter = new IntentFilter();
       intentFilter.addAction(Intent.ACTION_HEADSET_PLUG);
+      intentFilter.addAction(BluetoothHeadset.ACTION_CONNECTION_STATE_CHANGED);
       this.receiver = new BroadcastReceiver() {
           @Override
           public void onReceive(Context context, Intent intent) {
-              if (intent.getAction().equals(Intent.ACTION_HEADSET_PLUG)) {
-                  int state = intent.getIntExtra("state", -1);
-                  switch (state) {
-                  case 0:
-                      Log.d(LOG_TAG, "Headset is unplugged");
-                      mCachedWebView.sendJavascript("cordova.require('cordova-plugin-headsetdetection.HeadsetDetection').remoteHeadsetRemoved();");
-                      break;
-                  case 1:
-                      Log.d(LOG_TAG, "Headset is plugged");
-                      mCachedWebView.sendJavascript("cordova.require('cordova-plugin-headsetdetection.HeadsetDetection').remoteHeadsetAdded();");
-                      break;
-                  default:
-                      Log.d(LOG_TAG, "I have no idea what the headset state is");
-                  }
+              int status = getConnectionStatus(intent.getAction(), intent);
+
+              if (status == CONNECTED) {
+                Log.d(LOG_TAG, "Headset is connected");
+                mCachedWebView.sendJavascript("cordova.require('cordova-plugin-headsetdetection.HeadsetDetection').remoteHeadsetAdded();");
+              } else if (status == DISCONNECTED) {
+                Log.d(LOG_TAG, "Headset is disconnected");
+                mCachedWebView.sendJavascript("cordova.require('cordova-plugin-headsetdetection.HeadsetDetection').remoteHeadsetRemoved();");
+              } else {
+                Log.d(LOG_TAG, "Headset state is unknown: ${status}");
```

Comment:
Oh if only this was Kotlin. Unfortunately, you cannot do variables inside of strings like this in java. This will need to be `"Headset state is unknown: " + status;`

Reply:
ack! I'll get that changed. :)

Reply:
Fixed!